src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -158,6 +175,9 @@ private async Task AcceptClientsAsync()
                         continue;
                     }
 
+                    if (this.CloseClientNotWhiteListedDuringIBD(tcpClient))
```

Comment:
*Close client not whitelisted during IBD* (`CloseClientNotWhiteListedDuringIBD()`) is readable.  The developer will be able to understand the intent of this function, without having to read extra comments.

Closing clients is really a separate concern to *accepting clients* (`AcceptClientsAsync()`), so thought it would sit better in a separate function.

Finally, `AcceptClientsAsync()` would start to smell if we start to bloat it with more code.

Reply:
How is that

![1](https://user-images.githubusercontent.com/10259355/44535038-d0854180-a701-11e8-9ad2-acfd84583a61.png)

more readable than that

![2](https://user-images.githubusercontent.com/10259355/44535048-d549f580-a701-11e8-8f6a-83a1ad311b01.png)

?

Reply:
As in the method name explains what's going on without the need to read the code implementation.

This 

![image](https://user-images.githubusercontent.com/16000683/44537335-6ca94b80-a6f6-11e8-8a8f-3bdf0345c15c.png)

Is more  readable then this

![image](https://user-images.githubusercontent.com/16000683/44537357-7af76780-a6f6-11e8-8296-f2e56719c379.png)

Reply:
I suppose when you read the code you read it to improve or find a bug. In both cases you tend to read it all and not skip implementation.

Fox example from code like: 
```
DoSomething();

if (IsSomething())
    DoAnotherThing()
```

I can get the intention but I have no idea if it works as expected. 
In this code I won't find a bug until I check implementation of those methods. 

So we really should compare the pictures I've sent.

Reply:
how about 
```
                     if (!this.AllowClientConnection(tcpClient))
                     {
                         this.logger.LogTrace("Connection from client '{0}' was rejected and will be closed.", tcpClient.Client.RemoteEndPoint);
                         tcpClient.Close();
                         continue;
                     }
```
as a compromise? such a method could include the condition on line 171

Reply:
ok